debug_module.tex

```diff
@@ -68,8 +68,9 @@ \section{Reset Control} \label{reset}
 Exactly what is affected by this reset is implementation dependent, as long as
 it is possible to debug programs from the first instruction executed.
 The Debug Module's own state and registers should only be
```

Comment:
Perhaps, and I agree that the phrase "not recommended global reset" is awkward.
This paragraph appeared to enumerate all the sources of reset, and the omission of the global reset was somewhat confusing among my colleagues.
Also, there was some uncertainty in how strictly to interpret certain words (like "should", and "and"), since the paragraph uses the word ****and**** in a set union sense, rather than the boolean sense.
Maybe add the sentence "Resetting the DM from other implementation defined sources is permitted by not recommended."  Or just drop this change, this discussion is helpful clarification for me.

Reply:
I read the "and" as it was intended but I see how there could be confusion.  How about taking advantage of the fact that dmcontrol=0 on reset:
> The Debug Module’s own state and registers should only be reset while dmactive in dmcontrol is 0.

Or maybe something like:
> The Debug Module’s own state and registers should only be reset while dmactive in dmcontrol is 0 (including at power-up).

Reply:
I think I mostly understand what you are trying to say, but the wording is a little ambiguous.  Does the "should" apply to the debug module implementation or the thing talking to the debug module (i.e., the debugger)?

Do you mean:
- The debugger should only assert a reset of the debug module when the debugger has set dmactive in dmcontrol to 0, or
- In response to a request to reset the debug module, the debug module should only perform a reset of state when dmactive in dmcontrol is 0.

I think the first interpretation would be preferred, because of simpler hardware.  I would be happy with wording that clearly conveys that meaning.